### DISCUSSION TITLE
Usage of CoordinateSequenceFilter

### DISCUSSION THREAD

**Discussion by FObermaier** [locationtech/jts#842] [Q&A] [answered]
I want to test the `CoordinateSequence`s of geometries to see if they meet a certain condition. I understand that I can use `CoordinateSequenceFilter` for that, but in case of a `Polygon` it would be sufficient to check just the sequence of the shell / exterior ring. 

Can I prevent checking interior rings (for performance reasons)? If not, is there another filter class that would be more suitable?

**Comment by dr-jts** [comment, marked as answer]
I don't think there's any JTS filters which avoid checking interior rings.  You'll have to write your own iterator.
